File: 2018/12-Dec/12-22-multiple-inheritance/java/Main.java
/**
 * Test multiple inheritance in Java
 * @author Andrew Jarombek
 * @since 12/17/2018
 */

public class Main {
    public static void main(String... args) {
        var balsam = new BalsamFir(7, 2);

        assert balsam.isChristmasTree();
        assert balsam.leafPersistence();
        assert balsam.height().equals("(7, 2)");
    }
}

File: 2018/12-Dec/12-22-multiple-inheritance/php/main.php
<?php
/**
 * Test out using Traits for 'multiple inheritance' in PHP
 * User: Andrew Jarombek
 * Date: 12/18/2018
 */

include 'Tree.php';
include 'ChristmasTree.php';
include 'EvergreenTree.php';
include 'BalsamFir.php';

$tree_height = ["feet" => 7, "inches" => 2];

$balsam = new BalsamFir($tree_height);

print_r($balsam);

assert($balsam->type() == "Evergreen");
assert($balsam->height() == "7' 2\"");
assert($balsam->holidayTree);
